php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -1500,9 +1500,10 @@ abstract class NDB_BVL_Instrument extends NDB_Page
     }
 
     /**
-     * Determines what the data entry completion progress percentage is
+     * Determines what pecentage of the data entry has been completed, based on the
+     * number of required fields having data
```

Comment:
@driusan is this better?

Reply:
How is a "required field" determined?

Reply:
im not sure. i used the same "required fields" as used in the `_detemineDataEntryCompletionStatus()` function, which looks at $this->_requiredElements.

the function i'm proposing in this PR grabs the exact same data as in `_detemineDataEntryCompletionStatus()`. but instead of a boolean 'Complete' or 'Incomplete' , it returns the percentage of completeness. the results are the same, just with more detail.

i don't understand why this is sketchy if there's already a pretty important function that we're already using to determine data entry completeness that calculates via the same way

Reply:
The `_determineDataEntryCompletionStatus` function is only to determine if all data entry is complete. It's only written to assume a binary "Yes" or "No" answer. It requires that there is at least one field in `_requiredElements` per page in order to force the triggering of rules (which prevents saving the whole page), but not all required elements are in `_requiredElements`.

Reply:
hmmm so in that case, isn't it also a problem that data entry is being set as 'Complete' even though not all the required elements have been answered, and thus is not actually complete? i don't understand how `_determineDataEntryCompletionStatus` determines if all entry is complete if it only looks at the one field in `_requiredElements` and not all required elements in the instrument

Reply:
No, it can't be set to 'Complete' because an invalid rule (XINRule, QuickForm rule, whatever) prevents the page from being saved if anything is invalid on the page.

Reply:
I'm not saying there's anything wrong with using the field, I'm saying the comment needs to be clear about what it considers 'Complete Percentage' to avoid confusion in the future.

Reply:
ahh i see, so the function checks any leftover required fields not taken care of by the other rules. i'll need to look into this more. i can tag this PR as 'Needs Work'. ideally, all required fields should be considered in this completion progress function

Reply:
@driusan thanks for explaining this!

Reply:
I don't think it necessarily needs work if it's clear, but in most cases it's probably giving you the percentage of pages complete, not the percentage of questions.

Reply:
okayy that sounds good. as long as for linst instruments, it considers at least one field per page (which I think it does), then i'm happy to keep it this way

Reply:
this is not working as expected for LINST instruments unfortunately. the test instrument i have returns 100% even if all fields are empty